### PR TITLE
update CVE-2006-5752

### PR DESCRIPTION
**Pull request by kleenip**

### CODE REVIEW

File: cves/CVE-2006-5752.yml [hunk 1]
```diff
@@ -3,14 +3,15 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
```
Comment (scioffi) [inline cves/CVE-2006-5752.yml]: Please use just the CWE number (79) instead of words.

Comment (scioffi) [reply cves/CVE-2006-5752.yml]:
```suggestion
CWE: 79
```

File: cves/CVE-2006-5752.yml [hunk 2]
```diff
@@ -48,14 +53,19 @@ fixes_vcc_instructions: |
...
+    note:  
+  - commit: 15d69599aaf36251ee96f1bf6f8e6018b09be19a
+    note: in the history of the function from pointer to ap_rset_content_type
+          to the current function name of ap_set_content_type it seems
+          that the vulnerability has persisted. However, it seems that due 
+          to the reproduction steps requiring a browser that has charset
+          detection functionality, it most likely was not an oversight from
+          the beginning
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (scioffi) [inline cves/CVE-2006-5752.yml:63]: I like this CVE. 1 upvote from me please!

File: cves/CVE-2006-5752.yml (hunk 1 above)
Comment (andymeneely) [reply cves/CVE-2006-5752.yml]: Also, this is probably breaking your build (yml doesn't like colons here)

File: cves/CVE-2006-5752.yml [hunk 3]
```diff
@@ -48,14 +53,19 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml:53]: I couldn't put this higher, but is there any mention in the fix commits for the bug number? It seems odd for organizational sake.

File: cves/CVE-2006-5752.yml [hunk 4]
```diff
@@ -48,14 +53,19 @@ fixes_vcc_instructions: |
...
-     note:
+     note: specified a charset when setting content type
 vccs:
   - commit: dba4852ebca38b6c58f7600e4ef0d57bd9c81a8f
-    note:
-  - commit:
-    note:
+    note:  
+  - commit: 15d69599aaf36251ee96f1bf6f8e6018b09be19a
+    note: in the history of the function from pointer to ap_rset_content_type
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]: This note may be more suited for putting in in the mistake section.

File: cves/CVE-2006-5752.yml [hunk 5]
```diff
@@ -77,14 +87,14 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer:
+  answer: no
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]: Suggest adding an actual answer here and changing "no" to "false"

File: cves/CVE-2006-5752.yml [hunk 6]
```diff
@@ -96,11 +106,12 @@ discovered:
...
-  date:
-  automated:
-  google:
-  contest:
+  answer: The vulnerability was reported by Stefan Esser, the founder of
+          Hardened PHP and security expert. 
+  date: Jun 26 2007 12:00AM
+  automated: false
+  google: false
+  contest: nil
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]: Change this to false.

File: cves/CVE-2006-5752.yml [hunk 7]
```diff
@@ -109,7 +120,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer:
+  answer: the subsystem responsible for the displaying diagnostic information
+          of the server itself
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml:118]: Write the singular name of the subsystem in the name portion.

File: cves/CVE-2006-5752.yml [hunk 8]
```diff
@@ -168,8 +180,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: yes
+    note: xss vulnerability
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]: Could you expand on how this applies, more than just writing "xss vulnerability"?

File: cves/CVE-2006-5752.yml [hunk 9]
```diff
@@ -201,4 +214,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: It seems that emerging technology has a definite chance to break old code.
+          looking through previous iterations of the code, specifying a charset seemed
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]:
```suggestion
          Looking through previous iterations of the code, specifying a charset seemed
```

File: cves/CVE-2006-5752.yml [hunk 10]
```diff
@@ -201,4 +214,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: It seems that emerging technology has a definite chance to break old code.
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml]: Could you be a little more specific on what the mistake that the team made? Also is there something that we can learn from this vulnerability?

File: cves/CVE-2006-5752.yml [hunk 11]
```diff
@@ -109,7 +120,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer:
+  answer: the subsystem responsible for the displaying diagnostic information
+          of the server itself
   name:
 
 interesting_commits:
```
Comment (Dmj1097) [inline cves/CVE-2006-5752.yml:121]: Is there any interesting commits between the vulnerability was introduced vs when it was fixed? If not what went on during that time?

**Comment by Dmj1097** [comment]
Updated technology  vulnerabilities are always somewhat interesting to see, I give this 2 upvotes.

File: cves/CVE-2006-5752.yml (hunk 11 above)
Comment (Ksxerity) [inline cves/CVE-2006-5752.yml:121]: You should write a detailed description about what was happening between commits if there were not interesting commits to comment on.

File: cves/CVE-2006-5752.yml (hunk 8 above)
Comment (Ksxerity) [inline cves/CVE-2006-5752.yml]: This document may be viewed by people who do not know about cross site scripting so you should include a definition.

File: cves/CVE-2006-5752.yml [hunk 12]
```diff
@@ -168,8 +180,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: yes
```
Comment (Ksxerity) [inline cves/CVE-2006-5752.yml]: Should be "true" instead of "yes"

File: cves/CVE-2006-5752.yml [hunk 13]
```diff
@@ -185,9 +197,10 @@ lessons:
   yagni:
     applies:
     note:
-  complex_inputs:
-    applies:
-    note:
+  complex_inputs: 
+    applies: yes
```
Comment (Ksxerity) [inline cves/CVE-2006-5752.yml]: Should be "true" instead of "yes"

File: cves/CVE-2006-5752.yml [hunk 14]
```diff
@@ -48,14 +53,19 @@ fixes_vcc_instructions: |
...
+     note: specified a charset when setting content type 
    - commit: 6f0c8008746b58eb0a59f59501163eea2921d03d
-     note:
+     note: specified a charset when setting content type
 vccs:
   - commit: dba4852ebca38b6c58f7600e4ef0d57bd9c81a8f
-    note:
-  - commit:
-    note:
+    note:
```
Comment (Ksxerity) [inline cves/CVE-2006-5752.yml]: I'm assuming that this vcc was already provided for you, but you should still leave a note about why it was listed.

File: cves/CVE-2006-5752.yml (hunk 3 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml:53]: There is only mention of the CVE id.

File: cves/CVE-2006-5752.yml (hunk 8 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: expanded note.

File: cves/CVE-2006-5752.yml (hunk 7 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml:118]: added subsystem name.

File: cves/CVE-2006-5752.yml (hunk 6 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: changed.

File: cves/CVE-2006-5752.yml (hunk 4 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: moved to mistakes.

File: cves/CVE-2006-5752.yml (hunk 5 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: added answer.

File: cves/CVE-2006-5752.yml (hunk 9 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: fixed.

File: cves/CVE-2006-5752.yml (hunk 10 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: reformatted the answer to include mistake and lessons header, as well as notes.

File: cves/CVE-2006-5752.yml (hunk 8 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: added explanation.

File: cves/CVE-2006-5752.yml (hunk 12 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: changed.

File: cves/CVE-2006-5752.yml (hunk 13 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: changed.

File: cves/CVE-2006-5752.yml (hunk 14 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml]: removed vcc as it did not have any relation to the vulnerability introduced.

File: cves/CVE-2006-5752.yml (hunk 11 above)
Comment (kleenip) [reply cves/CVE-2006-5752.yml:121]: there were no commits that edited the line of code since the very beginning. updated to reflect this.

Comment (kleenip) [reply cves/CVE-2006-5752.yml:121]: there were no commits that edited the line of code since the very beginning. updated to reflect this.

**Pull request merged**